### PR TITLE
clang-tidy: convert to algorithms

### PR DESCRIPTION
**Pull request by neheb**
Found with: readability-use-anyofallof

### CODE REVIEW

File: taglib/ape/apetag.cpp [hunk 1]
```diff
@@ -65,13 +65,7 @@ namespace
...
     }
 
-    const String upperKey = String(key).upper();
-    for(auto k : invalidKeys) {
-      if(upperKey == k)
-        return false;
-    }
-
-    return true;
+    return std::none_of(invalidKeys.begin(), invalidKeys.end(), [upperKey = String(key).upper()](auto k) { return upperKey == k; });
```
Comment (sbooth) [inline taglib/ape/apetag.cpp:68]: Is the predicate `upperKey = String(key).upper()` evaluated for each iteration?

Comment (neheb) [reply taglib/ape/apetag.cpp:68]: no, only once. That's why it's in the capture list.

Comment (neheb) [reply taglib/ape/apetag.cpp:68]: btw to clarify, I need to use a lambda here because of ==. String needs to be on the left hand side.

Comment (sbooth) [inline taglib/ape/apetag.cpp:68]: Would `const auto &k` make more sense in the parameter list? Most of the other suggested lambdas use const references for objects so I wonder what is different about this one.

File: taglib/mp4/mp4atom.cpp [hunk 2]
```diff
@@ -97,19 +97,12 @@ MP4::Atom::Atom(File *file)
...
-          if(nextSize == child) {
-            // meta is not a full atom (i.e. not followed by version, flags). It
-            // is followed by the size and type of the first child atom.
-            metaIsFullAtom = false;
-            break;
-          }
-        }
+        // meta is not a full atom (i.e. not followed by version, flags). It
+        // is followed by the size and type of the first child atom.
+        bool metaIsFullAtom = std::none_of(metaChildrenNames.begin(), metaChildrenNames.end(), [nextSize = file->readBlock(8).mid(4, 4)](const auto &child) { return nextSize == child; });
```
Comment (sbooth) [inline taglib/mp4/mp4atom.cpp:105]: Should we prefer `auto metaIsFullAtom` here? It is inconsistent with the suggestion using `find_if` in textidentificationframe.

File: taglib/mpeg/id3v2/id3v2framefactory.cpp [hunk 3]
```diff
@@ -147,11 +147,9 @@ std::pair<Frame::Header *, bool> FrameFactory::prepareFrameHeader(
   }
 #endif
 
-  for(auto it = frameID.cbegin(); it != frameID.cend(); it++) {
-    if( (*it < 'A' || *it > 'Z') && (*it < '0' || *it > '9') ) {
-      delete header;
-      return {nullptr, false};
-    }
+  if(std::any_of(frameID.begin(), frameID.end(), [](auto c) { return (c < 'A' || c > 'Z') && (c < '0' || c > '9'); })) {
```
Comment (sbooth) [inline taglib/mpeg/id3v2/id3v2framefactory.cpp:150]: The original loop uses `cbegin` and `cend`. Is there an advantage to using the const versions?

File: taglib/riff/riffutils.h [hunk 4]
```diff
@@ -44,13 +44,7 @@ namespace TagLib
...
           return false;
 
-        for(auto it = name.begin(); it != name.end(); ++it) {
-          const int c = static_cast<unsigned char>(*it);
-          if(c < 32 || 127 < c)
-            return false;
-        }
-
-        return true;
+        return std::none_of(name.begin(), name.end(), [](char c) { return c < 32 || 127 < c; });
```
Comment (sbooth) [inline taglib/riff/riffutils.h:47]: Is `char` the correct type for `c`?

File: taglib/toolkit/tstring.cpp [hunk 5]
```diff
@@ -510,20 +510,12 @@ String String::stripWhiteSpace() const
 
 bool String::isLatin1() const
 {
-  for(ConstIterator it = begin(); it != end(); ++it) {
-    if(*it >= 256)
-      return false;
-  }
-  return true;
+  return std::none_of(this->begin(), this->end(), [](auto c) { return c >= 256; });
```
Comment (sbooth) [inline taglib/toolkit/tstring.cpp:513]: I have the same question as above about using const vs non-const iterators.

File: taglib/ape/apetag.cpp (hunk 1 above)
Comment (neheb) [reply taglib/ape/apetag.cpp:68]: auto vs const auto& boils down to whether or not the parameter is trivially copyable. No real difference except the former being shorter.

File: taglib/mp4/mp4atom.cpp (hunk 2 above)
Comment (neheb) [reply taglib/mp4/mp4atom.cpp:105]: find_if and none_of have different return values. auto offers no advantage here.

File: taglib/mpeg/id3v2/id3v2framefactory.cpp (hunk 3 above)
Comment (neheb) [reply taglib/mpeg/id3v2/id3v2framefactory.cpp:150]: Not really. This piece of code was actually transformed into a for range loop first which is incompatible with cbegin and cend.

File: taglib/mp4/mp4atom.cpp (hunk 2 above)
Comment (sbooth) [reply taglib/mp4/mp4atom.cpp:105]: I understand that, the question I'm raising is whether stylistically we should prefer the use of `auto`. It seems to be the recommendation in https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines

File: taglib/riff/riffutils.h (hunk 4 above)
Comment (neheb) [reply taglib/riff/riffutils.h:47]: Original code casts it to unsigned. Maybe it'd be good to do that here too.

**Comment by neheb** [comment]
Updated

**Pull request merged**